src/Microsoft.DotNet.ProjectModel/WorkspaceContext.cs

```diff
@@ -251,17 +330,9 @@ private FileModelEntry<LockFile> AddLockFileEntry(string projectDirectory, FileM
             {
                 currentEntry.Reset();
 
-                foreach (var framework in project.GetTargetFrameworks())
-                {
-                    var builder = new ProjectContextBuilder()
-                        .WithProjectResolver(path => GetProject(path).Model)
-                        .WithLockFileResolver(path => GetLockFile(path))
-                        .WithProject(project)
-                        .WithTargetFramework(framework.FrameworkName)
-                        .AsDesignTime();
-
-                    currentEntry.ProjectContexts.Add(builder.Build());
-                }
+                var builder = InitializeProjectContextBuilder(project);
```

Comment:
why do we need the `builder` as an intermediate here? can we have the function return all contexts together?